specification/devcenter/DevCenter/shared/models.tsp

```diff
@@ -6,53 +6,38 @@ import "../main.tsp";
 using Azure.Core;
 using TypeSpec.Rest;
 using TypeSpec.Http;
+using TypeSpec.Versioning;
 
 namespace DevCenterService;
 
-@doc("Indicates whether operation status is running, completed, canceled or failed.")
-@lroStatus
-union OperationStatusValue {
-  string,
-
-  @doc("Operation is in progress")
-  Running: "Running",
-
-  @doc("Operation is completed with success")
-  @lroSucceeded
-  Completed: "Completed",
-
-  @doc("Operation was canceled")
-  Canceled: "Canceled",
-
-  @doc("Operation failed")
-  Failed: "Failed",
-}
-
-@doc("The current status of an async operation")
+@doc("The current status of an async operation.")
 @resource("operationstatuses")
 @parentResource(Project)
 model OperationStatus {
   @doc("Fully qualified ID for the operation status.")
+  @maxLength(36)
+  @minLength(36)
+  @pattern("^[a-zA-Z0-9]{8}-([a-zA-Z0-9]{4}-){3}[a-zA-Z0-9]{12}$")
   @key("operationId")
   @visibility("read")
```

Comment:
glad we add the constraints here because we have just catch an error! The build is failing because @key is under the wrong property! Id is of type "/projects/{project}/operationStatuses/{operationId}", is not the operationId. Name is the property that has the operation id. Lets move all that over there

Reply:
lets make `Name` a  `uuid` type, setting the `@key("operationId")` there. I think if we set that as uuid we might not even need to add the constraints (check if the swagger will show the constraints. if not we add them)